### PR TITLE
cargo-workspaces: init at 0.2.35

### PR DESCRIPTION
**Pull request by macalinao**
###### Description of changes

Adds cargo-workspaces to nixpkgs.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dit7ya** [comment]
There is also https://github.com/NixOS/nixpkgs/pull/181792

File: pkgs/development/tools/rust/cargo-workspaces/default.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+, openssl
+, zlib
+, stdenv
+, darwin
+, libssh2
+, libgit2
+}:
+
+with rustPlatform;
+
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rust/cargo-workspaces/default.nix:14]:
```suggestion
```
Please prefix the variables instead.

File: pkgs/development/tools/rust/cargo-workspaces/default.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  LIBGIT2_SYS_USE_PKG_CONFIG = true;
+  LIBSSH2_SYS_USE_PKG_CONFIG = true;
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ openssl zlib libssh2 libgit2 ] ++ (
+    lib.optionals stdenv.isDarwin
+      (with darwin.apple_sdk.frameworks;
+      ([ IOKit Security CoreFoundation AppKit ]
+        ++ (lib.optionals stdenv.isAarch64 [ System ])))
+  );
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rust/cargo-workspaces/default.nix:37]:
```suggestion
  nativeBuildInputs = [ pkg-config ];

  buildInputs = [ openssl zlib libssh2 libgit2 ]
    ++ lib.optionals stdenv.isDarwin [ AppKit CoreFoundation IOKit Security System ];
```

File: pkgs/development/tools/rust/cargo-workspaces/default.nix [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  strictDeps = true;
+
+  meta = with lib; {
+    description =
+      "A tool for managing cargo workspaces and their crates, inspired by lerna";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rust/cargo-workspaces/default.nix:42]:
```suggestion
    description = "A tool for managing cargo workspaces and their crates, inspired by lerna";
```

File: pkgs/development/tools/rust/cargo-workspaces/default.nix [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    longDescription = ''
+      A tool that optimizes the workflow around cargo workspaces with
+      git and cargo by providing utilities to version, publish, execute
+      commands and more.
+    '';
+    homepage = "https://github.com/pksunkara/cargo-workspaces";
+
+    license = licenses.mit;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rust/cargo-workspaces/default.nix:50]:
```suggestion
    homepage = "https://github.com/pksunkara/cargo-workspaces";
    license = licenses.mit;
```

**Comment by macalinao** [comment]
any idea what it would take to get this merged? No rush-- just curious if I've missed anything. Thanks!

**Comment by RaitoBezarius** [comment]
> any idea what it would take to get this merged? No rush-- just curious if I've missed anything. Thanks!

Did you coordinate with @cjab for this PR as he put himself as a maintainer of this package?

**Comment by macalinao** [comment]
@cjab want to be a maintainer on this?

**Comment by cjab** [comment]
No worries @macalinao it's all you! 😄 It looks like your package is a bit more robust than mine. I'll go ahead and close my PR.

**Pull request merged**